compiler/typecheck/TcMType.hs

```diff
@@ -2146,10 +2146,17 @@ tidySigSkol env cx ty tv_prs
         (env', tv') = tidy_tv_bndr env tv
 
     tidy_ty env (FunTy w arg res)
-      = FunTy w (tidyType env arg) (tidy_ty env res)
+      = FunTy (tidyMult env w) (tidyType env arg) (tidy_ty env res)
 
     tidy_ty env ty = tidyType env ty
 
+    tidyMult _   Zero = Zero
+    tidyMult _   One = One
+    tidyMult _   Omega = Omega
+    tidyMult env (MultThing ty) = MultThing (tidy_ty env ty)
+    tidyMult env (MultAdd x y) = MultAdd (tidyMult env x) (tidyMult env y)
+    tidyMult env (MultMul x y) = MultMul (tidyMult env x) (tidyMult env y)
```

Comment:
There's an inordinate amount of code like this. We should write a traversal-like-thing for `GMult` (not a `Traversable` instance because of restrictions)

```haskell
-- | 'traverseMult f' is only a proper if 'f' doesn't change its argument into something which 'fromMult' would reify.
traverseMult :: forall f a b. (Applicative f, Multable a, Multable b) => (a -> f b) -> (GMult a -> f (GMult b)
```

(a refactoring for another commit)